azext_edge/edge/providers/base.py

```diff
@@ -366,6 +366,19 @@ def delete_namespaced_custom_object(
             raise RuntimeError(error_msg)
 
 
+def get_namespaced_configmap(namespace: str, cm_name: str) -> dict:
```

Comment:
Why not use https://github.com/Azure/azure-iot-ops-cli-extension/blob/dev/azext_edge/edge/providers/k8s/config_map.py#L16 ?

Also if you are going to add new k8s resource fetch functions, please put them in the k8s subpackage like above.